LLama/Native/LLamaLogLevel.cs

```diff
@@ -4,29 +4,41 @@
 namespace LLama.Native
 {
     /// <summary>
-    /// Severity level of a log message
+    /// Severity level of a log message. This enum should always be aligned with
+    /// the one defined on llama.cpp side at
+    /// https://github.com/ggerganov/llama.cpp/blob/0eb4e12beebabae46d37b78742f4c5d4dbe52dc1/ggml/include/ggml.h#L559
     /// </summary>
     public enum LLamaLogLevel
     {
         /// <summary>
-        /// Logs that highlight when the current flow of execution is stopped due to a failure.
+        /// Logs are never written.
         /// </summary>
-        Error = 2,
+        None = 0,
 
         /// <summary>
-        /// Logs that highlight an abnormal or unexpected event in the application flow, but do not otherwise cause the application execution to stop.
+        /// Logs that are used for interactive investigation during development.
         /// </summary>
-        Warning = 3,
+        Debug = 1,
 
         /// <summary>
         /// Logs that track the general flow of the application.
         /// </summary>
-        Info = 4,
+        Info = 2,
 
         /// <summary>
-        /// Logs that are used for interactive investigation during development.
+        /// Logs that highlight an abnormal or unexpected event in the application flow, but do not otherwise cause the application execution to stop.
         /// </summary>
-        Debug = 5,
+        Warning = 3,
+
+        /// <summary>
+        /// Logs that highlight when the current flow of execution is stopped due to a failure.
+        /// </summary>
+        Error = 4,
+        
+        /// <summary>
+        /// Continue log level is equivalent to None in the way it is used in llama.cpp.
+        /// </summary>
+        Continue = 5,
     }
 
     internal static class LLamaLogLevelExtensions
```

Comment:
```suggestion
    internal static class LLamaLogLevelExtensions
    {
        [ThreadStatic] private static LogLevel _previous;
        
        public static LogLevel ToLogLevel(this LLamaLogLevel llama)
        {
            _previous = (llama) switch
            {
                LLamaLogLevel.None => LogLevel.None,
                LLamaLogLevel.Debug => LogLevel.Debug,
                LLamaLogLevel.Info => LogLevel.Information,
                LLamaLogLevel.Warning => LogLevel.Warning,
                LLamaLogLevel.Error => LogLevel.Error,
                LLamaLogLevel.Continue => _previous,
                _ => throw new ArgumentOutOfRangeException(nameof(llama), llama, null)
            };
            return _previous;
    }
```

Reply:
(Note that I haven't tested this, but hopefully it illustrates what I mean)

Reply:
Got it, good idea